dali/operators/reader/loader/recordio_loader.h

```diff
@@ -100,10 +100,11 @@ class RecordIOLoader : public IndexedFileLoader {
       return;
     }
 
-    if (should_seek_) {
+    if (should_seek_ || next_seek_pos_ != seek_pos) {
       current_file_->Seek(seek_pos);
       should_seek_ = false;
     }
+    next_seek_pos_ = seek_pos + size;
```

Comment:
This is very brittle and may lead to very subtle and hard-to-detect errors - we have not moved the file pointer yet.

Reply:
This should be in lines:
120, 135
Also, it looks like line 141 we should reset next_seek_pos_ to 0   ?

Reply:
Done